plugins/preset-chart-xy/src/BoxPlot/legacy/transformProps.ts

```diff
@@ -22,7 +22,9 @@ import { RawBoxPlotDataRow, BoxPlotDataRow } from '../../components/BoxPlot/type
 export default function transformProps(chartProps: ChartProps) {
   const { width, height, datasource, formData, queryData } = chartProps;
   const { verboseMap = {} } = datasource;
-  const { colorScheme, groupby, metrics } = formData;
+  const { colorScheme } = formData;
+  const groupby = formData.groupby as string[];
+  const metrics = formData.metrics as string[];
```

Comment:
Maybe expand the `ChartProps` like [BigNumber](https://github.com/apache-superset/superset-ui/blob/20e70244548f167e206a6f822569305321513ec5/plugins/legacy-preset-chart-big-number/src/BigNumber/transformProps.ts#L41-L62) for consistency?

I'm wondering whether we can add these generic `formData` properties to the global `formData`, too. They would go hand-in-hand with the shared controls.

Reply:
Great idea.